lib/parse/client.rb

```diff
@@ -69,6 +69,11 @@ def request(uri, method = :get, body = nil, query = nil, content_type = nil)
       end
 
       @session.send(method, uri, query || body || {}, headers).body
+
+    # Don't leak our internal libraries to our clients.
+    # Extend this list of exceptions as needed.
+    rescue Faraday::Error::ConnectionFailed, Faraday::Error::TimeoutError
+      raise Parse::ConnectionError
```

Comment:
@rhymes Thanks for the feedback! Just to make sure I understand you, are you saying that you would like to access the original exception? In Ruby 2.1 you can call `Exception#cause`: http://devblog.avdi.org/2013/12/25/exception-causes-in-ruby-2-1/

How do you feel about that?